### PR TITLE
Select nuget version based on msbuild version

### PR DESCRIPTION
**Pull request by phil-hodgson**

### CODE REVIEW

File: Tasks/Common/packaging-common/nuget/NuGetToolGetter.ts [hunk 1]
```diff
@@ -98,11 +99,40 @@ export async function getNuGet(versionSpec: string, checkLatest?: boolean, addNu
...
+                taskLib.debug('Found msbuild version: ' + version);
+            }
+        });
+        await getVersionTool.exec();
+    }
+    return version;
+}
+
+export async function getMSBuildVersion(): Promise<semver.SemVer> {
+    const version = await getMSBuildVersionString();
```
Comment (satbai) [inline Tasks/Common/packaging-common/nuget/NuGetToolGetter.ts:121]: Add verbose logging here to show the msbuild version.

File: make-util.js [hunk 2]
```diff
@@ -392,6 +392,9 @@ var downloadFile = function (url) {
 }
 exports.downloadFile = downloadFile;
 
+// string hashing from https://stackoverflow.com/questions/7616461/generate-a-hash-from-string-in-javascript
+const hashCode = s => s.split('').reduce((a,b)=>{a=((a<<5)-a)+b.charCodeAt(0);return a&a},0);
```
Comment (satbai) [inline make-util.js:396]: can you add a comment why this is necessary for those who don't have context

File: Tasks/Common/packaging-common/nuget/NuGetToolGetter.ts [hunk 3]
```diff
@@ -98,11 +99,41 @@ export async function getNuGet(versionSpec: string, checkLatest?: boolean, addNu
...
+    return semver.coerce(version);
```
Comment (satbai) [inline Tasks/Common/packaging-common/nuget/NuGetToolGetter.ts:122]: is this going to fail if version is null?

Comment (johnterickson) [reply Tasks/Common/packaging-common/nuget/NuGetToolGetter.ts:122]: passes the null through

File: make-util.js (hunk 2 above)
Comment (johnterickson) [reply make-util.js]: removed

**Pull request merged**